doc/examples/parallel_planning/src/parallel_planning_main.cpp

```diff
@@ -0,0 +1,303 @@
+    {
+      planning_component_->setGoal(planning_query_request_.goal_constraints);
+    }
+    else
+      RCLCPP_ERROR(LOGGER, "Planning query request does not contain any goal constraints");
+  }
+
+  /// \brief Request a motion plan based on the assumption that a goal is set and print debug information.
+  void planAndPrint()
+  {
+    // Set start state as current state
+    planning_component_->setStartStateToCurrentState();
+
+    moveit_cpp::PlanningComponent::MultiPipelinePlanRequestParameters multi_pipeline_plan_request{
+      node_, { "ompl_rrtc", "pilz_lin", "chomp" }
+    };
+
+    auto plan_solution = planning_component_->plan(multi_pipeline_plan_request, &getShortestSolution);
+
+    // Check if PlanningComponents succeeded in finding the plan
+    if (plan_solution)
+    {
+      // Visualize the trajectory in Rviz
+      auto robot_model_ptr = moveit_cpp_->getRobotModel();
+      auto robot_start_state = planning_component_->getStartState();
+      auto joint_model_group_ptr = robot_model_ptr->getJointModelGroup(PLANNING_GROUP);
+
+      visual_tools_.publishTrajectoryLine(plan_solution.trajectory_, joint_model_group_ptr);
+      visual_tools_.trigger();
+    }
+
+    planning_component_->execute();
+
+    // Start the next plan
+    visual_tools_.prompt("Press 'next' in the RvizVisualToolsGui window to continue the demo");
+    visual_tools_.deleteAllMarkers();
+    visual_tools_.trigger();
+  }
+
+private:
+  std::shared_ptr<rclcpp::Node> node_;
+  std::shared_ptr<moveit_cpp::MoveItCpp> moveit_cpp_;
+  std::shared_ptr<moveit_cpp::PlanningComponent> planning_component_;
+  moveit_visual_tools::MoveItVisualTools visual_tools_;
+  moveit_msgs::msg::MotionPlanRequest planning_query_request_;
+};
+}  // namespace parallel_planning_example
+
+int main(int argc, char** argv)
+{
+  rclcpp::init(argc, argv);
+  rclcpp::NodeOptions node_options;
+  RCLCPP_INFO(LOGGER, "Initialize node");
+
+  node_options.automatically_declare_parameters_from_overrides(true);
+  rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("parallel_planning_example", "", node_options);
+
+  rclcpp::executors::MultiThreadedExecutor executor;
+  executor.add_node(node);
+  std::thread([&executor]() { executor.spin(); }).detach();
+
+  parallel_planning_example::Demo demo(node);
+
+  /* Otherwise robot with zeros joint_states */
+  rclcpp::sleep_for(std::chrono::seconds(1));
+
+  if (!demo.loadPlanningSceneAndQuery())
+  {
+    rclcpp::shutdown();
+    return 0;
+  }
+
+  RCLCPP_INFO(LOGGER, "Starting MoveIt Tutorials...");
+
+  // // Experiment 1 - Short free-space motion
+  // // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  RCLCPP_INFO(LOGGER, "Experiment 1 - Short free-space motion");
+
+  demo.setJointGoal(0.0, -0.8144019900299497, 0.0, -2.6488387075338133, 0.0, 1.8344367175038623, 0.7849999829891612);
+  demo.planAndPrint();
+
+  // // Experiment 2 - Long motion with collisions
+  // // ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Comment:
May say in these 2 experiments which planner you expect to be the winners?

I assume Pilz for 1, OMPL for 2